core/blocktree/blocktree.go

```diff
@@ -33,30 +34,31 @@ type Hash = common.Hash
 
 // BlockTree represents the current state with all possible blocks
 type BlockTree struct {
-	head            *node
+	head            *Node
 	leaves          leafMap
-	finalizedBlocks []*node
+	finalizedBlocks []*Node
 	Db              *polkadb.BlockDB
 }
 
 // NewBlockTreeFromGenesis initializes a blocktree with a genesis block.
 func NewBlockTreeFromGenesis(genesis types.Block, db *polkadb.BlockDB) *BlockTree {
-	head := &node{
-		hash:     genesis.Header.Hash,
-		number:   genesis.Header.Number,
-		parent:   nil,
-		children: []*node{},
-		depth:    big.NewInt(0),
+	head := &Node{
+		Hash:        genesis.Header.Hash,
+		Number:      genesis.Header.Number,
+		parent:      nil,
+		children:    []*Node{},
+		depth:       big.NewInt(0),
+		ArrivalTime: uint64(time.Now().UnixNano() / 1e6), // set arrival time of genesis in milliseconds since unix epoch
```

Comment:
Maybe we should make this a common function (`func unixMilli`)? or at least a function in this package?